pyfr/util.py

```diff
@@ -152,7 +152,8 @@ def subclass_where(cls, **kwargs):
     k, v = next(iter(kwargs.items()))
 
     for s in subclasses(cls):
-        if hasattr(s, k) and getattr(s, k) == v:
+        if hasattr(s, k) and (getattr(s, k) == v or
+                              any(i == v for i in getattr(s, k))):
```

Comment:
This is probably not a good design; if only because strings are iterable.

Reply:
I'm still working on this. Everything else should be ok.

Reply:
So I think we should consider extending `subclass_where` to process multiple arguments so we could say `subclass_where(Base, name=name, po=po)`.  This should not be too much more work and would provide us with a simple means of differentiating schemes within a family.

Reply:
I am not sure. I don't think differentiating schemes within a family is the issue here. 

The main challenge is finding the DensePseudoStepper subclass for all dense families. Lets say we have 'rkopt-s7-px.txt', 'otherdensescheme-s5.txt',  'magic-s4-px.txt' which are all dense schemes. The subclass where need to match all these names 'rkopt', 'otherdensescheme' and 'magic' to the DensePseudoStepper class. This now can be done by adding them into

class DualDenseRKPseudoStepper(BaseDualPseudoStepper):
    pseudo_stepper_name = ['rkopt', 'otherdensescheme', 'magic-s4' ]

And having the modified sub_class where that can handle lists:

def subclass_where(cls, **kwargs):
    k, v = next(iter(kwargs.items()))

    for s in subclasses(cls):
        if hasattr(s, k):
            if getattr(s, k) == v:
                return s
            elif type(getattr(s, k)) == list and any(i == v for i
                                                     in getattr(s, k)):
                return s

    raise KeyError("No subclasses of {0} with cls.{1} == '{2}'"
                   .format(cls.__name__, k, v))

The constructor queries the polynomial order from the cfg and the right scheme within gets picked since multip passes p-order modifified cfgs to the pseudo-integrators.

Reply:
This is not a huge challenge.  All we need to do in an __init__ block somewhere is list all of the available dense schemes and then create (sub) classes for them using the type method.  This will register them with PyFR (indeed, it will not be able to tell the difference between a class whose coefficients are hard coded and one which reads them from disk).